packages/svelte2tsx/src/emitDts.ts

```diff
@@ -13,7 +13,20 @@ export async function emitDts(config: EmitDtsConfig) {
     const { options, filenames } = loadTsconfig(config, svelteMap);
     const host = await createTsCompilerHost(options, svelteMap);
     const program = ts.createProgram(filenames, options, host);
-    program.emit();
+
+    // EmitResult below doesn't contain them for some reason, so get them separately
+    const diagnostics = ts.getPreEmitDiagnostics(program);
```

Comment:
I'm wondering how much slower the emit is becoming through this, AFAICT TS will do some thorough code analyses there (which might also close #1835). Should we have this behind an option?

Reply:
The problem with rolling our custom warnings is that it doesn't contain information on where the bad import happened. I suppose we could set a flag if such files are found and only run the diagnostics if it's true?

Reply:
As for #1835, I don't think emitDts's diagnostics can be fully relied upon. It already gets a few false positives because the tsconfig passed to it doesn't line up with the generated by SvelteKit.